client/task-list/tasks/appearance.js

```diff
@@ -160,6 +156,7 @@ class Appearance extends Component {
 			method: 'POST',
 		} )
 			.then( ( response ) => {
+				clearTaskStatusCache();
```

Comment:
This is a clever way of making sure the data gets accurately updated, but there does seem to be a small disconnect between these data stores and it feels hacky that we need to clear the resolution cache on a different store to get this working.

@samueljseay I think this may be similar to what you were you discussing in your p2 post; do you have any thoughts here?

Reply:
🤔 I think regardless of cross store communication, the main issue again for me is complex async flows in components and by concerning the component with cache clearing we do just increase that coupling here. I think a code smell we have is where we put large amounts of store related code into the component as props and thats definitely not a redux-like approach. 

All that said, this is obviously a refactor born out of the most pragmatic path forward ❤️ . The only way to fix problems when you have our existing patterns is to continue importing more information from the store into the component, so I don't have any concerns here, except this component would be another one I'd want to spend some time refactoring later. 

cc @jeffstieler